lisa/secret.py

```diff
@@ -17,7 +17,14 @@
     r"\1***\2",
 )
 
-patterns = {"guid": PATTERN_GUID, "headtail": PATTERN_HEADTAIL}
+# https://xx.core.windows.net/vhds/CentOS.vhd?sp=r&st=xx%2012:10:45&5;CA%aEMpls3D
+# replace as https://xx.core.windows.net/vhds/CentOS.vhd***
+PATTERN_URL = (
+    re.compile(r"(https?://([-\w]+\.)+[-\w]+(/[-./\w]*)?)(\?[\w]+=[%&-:;=\w]*)$"),
```

Comment:
The pattern should support normal url too, so it doesn't need to check if it's a normal url or not.

`(https?://([-\w]+\.)+[-\w]+(/[-./\w]*)?\??)([\w]+=[%&-:;=\w]*)?$`

Reply:
If so, the normal url will also print as "https://xx.core.windows.net/vhds/CentOS.vhd***", but the "***" is not we want.

Reply:
The vhd*** is acceptable, and we know if there is no "?", it means no parameters. And it doesn't need to break the current code logic. The "replace" in secret is dangerous, it may cause the mask failed in some cases.